cms/static/js/views/assets.js

```diff
@@ -41,66 +37,81 @@ define(["jquery", "underscore", "gettext", "js/models/asset", "js/views/paging",
+                    tableBody = this.getTableBody();
+                    tableBody.empty();
+                    if (hasAssets) {
+                        assets.each(
+                            function(asset) {
+                                var view = new AssetView({model: asset});
+                                tableBody.append(view.render().el);
+                            }
+                        );
+                    }
+                    self.$('.assets-library').toggle(hasAssets);
+                    self.$('.no-asset-content').toggle(!hasAssets);
+                    return this;
+                },
+
+                getTableBody: function() {
+                    var tableBody = this.tableBody;
+                    if (!tableBody) {
+                        ViewUtils.hideLoadingIndicator();
+
+                        // Create the table
+                        this.$el.html(_.template(asset_library_template, {typeData: this.typeData}));
+                        tableBody = this.$('#asset-table-body');
+                        this.tableBody = tableBody;
+                        this.pagingHeader = new PagingHeader({view: this, el: $('#asset-paging-header')});
+                        this.pagingFooter = new PagingFooter({view: this, el: $('#asset-paging-footer')});
+                        this.pagingHeader.render();
+                        this.pagingFooter.render();
+
+                        // Hide the contents until the collection has loaded the first time
+                        this.$('.assets-library').hide();
+                        this.$('.no-asset-content').hide();
+                    }
+                    return tableBody;
+                },
 
-            getTableBody: function() {
-                var tableBody = this.tableBody;
-                if (!tableBody) {
+                onError: function() {
                     ViewUtils.hideLoadingIndicator();
-
-                    // Create the table
-                    this.$el.html(this.template({typeData: this.typeData}));
-                    tableBody = this.$('#asset-table-body');
-                    this.tableBody = tableBody;
-                    this.pagingHeader = new PagingHeader({view: this, el: $('#asset-paging-header')});
-                    this.pagingFooter = new PagingFooter({view: this, el: $('#asset-paging-footer')});
-                    this.pagingHeader.render();
-                    this.pagingFooter.render();
-
-                    // Hide the contents until the collection has loaded the first time
-                    this.$('.assets-library').hide();
-                    this.$('.no-asset-content').hide();
                 }
-                return tableBody;
+            }),
+
+            createPagingView: function() {
+                var pagingView = new this.PagingAssetView({
+                        collection: this.collection
+                    });
+                this.$el.append(pagingView.render().el);
```

Comment:
I vote for doing the render in the render method. I don't think it will get called over and over, and that is where it is intended that the rendering is done.